base/src/main/java/com/smartdevicelink/managers/lifecycle/BaseLifecycleManager.java

```diff
@@ -951,9 +961,17 @@ public void onSessionStarted(int sessionID, Version version, SystemInfo systemIn
 
             if (systemInfo != null && lifecycleListener != null) {
                 didCheckSystemInfo = true;
-                String address = session.getBluetoothMacAddress();
-                if (address != null && !address.isEmpty()) {
-                    saveVehicleType(address, systemInfo.getVehicleType());
+                List<TransportRecord> activeTransports = session.getActiveTransports();
+                if (activeTransports != null) {
+                    for (TransportRecord record: activeTransports) {
+                        if (record.getType() == TransportType.BLUETOOTH) {
+                            String address = record.getAddress();
+                            if (address != null && !address.isEmpty()) {
+                                saveVehicleType(address, systemInfo.getVehicleType());
+                            }
+                            break;
+                        }
+                    }
```

Comment:
@YaroslavLutsenko  I think this block of code is a good candidate for a separate function as you have the same code duplication above

Reply:
@AKalinich-Luxoft How about: 
`private void saveVehicleType(List<TransportRecord> activeTransports, VehicleType type) {
        if (activeTransports == null || activeTransports.isEmpty() || type == null) {
            DebugTool.logWarning(TAG, "Unable to save vehicle type");
            return;
        }

        for (TransportRecord record: activeTransports) {
            if (record.getType() == TransportType.BLUETOOTH) {
                String address = record.getAddress();
                if (address != null && !address.isEmpty()) {
                    saveVehicleType(address, type);
                }
                break;
            }
        }
    }`

Reply:
@AKalinich-Luxoft Thank you for your advice. I created function 'private void saveVehicleType(List<TransportRecord> activeTransports, VehicleType type)'